apps/dashboard/src/app/team/[team_slug]/(team)/~/settings/SettingsLayout.tsx

```diff
@@ -0,0 +1,62 @@
+"use client";
```

Comment:
This is not a new page - renamed layout.tsx to SettingsLayout.tsx

Reply:
Feels odd to put it inside components - This is basically the entire layout but had to move it to a new file to make it a client component

Reply:
does the whole component actually need to be a client component? if so can the whole settings layout be a client component? (layouts are allowed to be client components, too)

Reply:
Can't make the layout.tsx a client component because it's async. 

But the layout needs to be a client component because of useState usage - so had to pull out the UI to a separate non-async component that can be made a client component